app/scripts/lib/segment.js

```diff
@@ -20,8 +22,86 @@ const segmentNoop = {
 // provided a SEGMENT_WRITE_KEY. This also holds true for test environments and
 // E2E, which is handled in the build process by never providing the SEGMENT_WRITE_KEY
 // which process.env.IN_TEST is true
-export const segment = process.env.SEGMENT_WRITE_KEY
+const segment = process.env.SEGMENT_WRITE_KEY
   ? new Analytics(process.env.SEGMENT_WRITE_KEY, { flushAt })
   : segmentNoop
 
-export const METAMETRICS_ANONYMOUS_ID = '0x0000000000000000'
+/**
+ * Returns a function for tracking Segment events.
+ *
+ * @property {Function} getParticipateInMetrics - A function that returns
+ * whether the user participates in MetaMetrics.
+ * @property {Function} getMetaMaskVersion - A function for getting the current
+ * version of the MetaMask extension.
+ * @property {Function} getMetricsState - A function for getting state relevant
+ * to MetaMetrics/Segment.
+ */
+export function getTrackSegmentEvent (
+  getParticipateInMetrics,
+  getMetaMaskVersion,
+  getMetricsState,
+) {
+  const getVersion = () => {
+    const metamaskVersion = getMetaMaskVersion()
+    return process.env.METAMASK_ENVIRONMENT === 'production'
+      ? metamaskVersion
+      : `${metamaskVersion}-${process.env.METAMASK_ENVIRONMENT}`
+  }
+
+  const segmentContext = {
+    app: {
+      name: 'MetaMask Extension',
+    },
+    page: {
+      path: '/background-process',
+      title: 'Background Process',
+      url: '/background-process',
+    },
+    userAgent: window.navigator.userAgent,
+  }
+
+  /**
+   * Tracks a Segment event per the given arguments.
+   *
+   * @param {string} event - The event name.
+   * @param {string} category - The event category.
+   * @param {Object} [properties] - The event properties.
+   * @param {boolean} [anonymous] - `true` if the user's MetaMetrics id should
+   * not be included, and `false` otherwise. Default: `true`
+   */
+  return function trackSegmentEvent ({
+    event,
+    category,
+    properties = {},
+    anonymous = true,
+  }) {
+    if (!getParticipateInMetrics()) {
+      return
+    }
+
+    const { currentLocale, metaMetricsId } = getMetricsState()
+
+    const trackOptions = {
+      event: event || 'DEFAULT_SEGMENT_EVENT',
+      category: category || 'DEFAULT_SEGMENT_CATEGORY',
+      context: {
+        ...segmentContext,
+        app: {
+          ...segmentContext.app,
+          version: getVersion(),
+        },
+        locale: currentLocale.replace('_', '-'),
+      },
+      properties,
+    }
+
+    if (anonymous) {
+      trackOptions.anonymousId = METAMETRICS_ANONYMOUS_ID
+      trackOptions.excludeMetaMetricsId = true
```

Comment:
I don't think this property should be here, despite it being present in the pre-existing segment track options. The `excludeMetaMetricsId` property was only meant for internal use, to pass to our track function to tell it to exclude the id (e.g. what you're now using `anonymous` for)

Unless that has changed? cc @danjm @brad-decker

Reply:
no it shouldn't be sent to segment, its only for internal use. The analytics tool can derive this value from being tracked as an anonymous user with the 0x000 id.

Reply:
aed190c